### PR TITLE
Merge glow-h changes in internal repo to github master

### PR DESCRIPTION
**Pull request by arunm-git**
*Description*:

This commit brings over changes that were made internally at FB for glow-h.
* Most of the changes are in lib/Backends/Habana/* and are just replaying changes made internally. Hopefully reviewing those should be easy enough for @bertmaher or @rdzhabarov who made most of those changes.
* DeviceManagerTest changes are to parameterize by backend kind
* HabanaBackendTest change adds one test case which is currently disabled.

*Testing*:
ninja all && ninja test
Habana specific testing will be done internally.

### CODE REVIEW

File: lib/Backends/Habana/Habana.cpp [hunk 1]
```diff
@@ -307,6 +286,8 @@ class TensorHandle final {
   /// Get the underlying data buffer.
   void *getData() const { return buffer_; }
 
+  const std::string &getName() const { return name_; }
```
Comment (rdzhabarov) [inline lib/Backends/Habana/Habana.cpp]: we should add doxygen, but probably in fbcode first.

File: lib/Base/Tensor.cpp [hunk 2]
```diff
@@ -148,6 +148,8 @@ static void dumpGenericImpl(Handle<ElemTy> handle, llvm::raw_ostream &os,
   }
 
   os << "]\n";
+
+  os.flush();
```
Comment (rdzhabarov) [inline lib/Base/Tensor.cpp]: looks legit, but wondering how that sneaked in :)

File: lib/Onnxifi/onnxifiGlow.cpp [hunk 3]
```diff
@@ -59,6 +59,11 @@ GLOW_ONNXIFI_LIBRARY_FUNCTION_WRAPPER(onnxGetBackendIDs)(
 #else
   constexpr bool withCPU = false;
 #endif
+#ifdef GLOW_WITH_HABANA
```
Comment (rdzhabarov) [inline lib/Onnxifi/onnxifiGlow.cpp]: I think changes in this file are good to go, but we need to have a principled way of handling non open sourced backend in the onnxifi glow.

**Comment by rdzhabarov** [comment]
can you check the test failure:
```
../lib/Backends/OpenCL/OpenCL.cpp:168: failed assertion `kernel && "clCreateKernel Failed."'
```

does not look like just a spurious failure.

File: lib/Base/Tensor.cpp (hunk 2 above)
Comment (bertmaher) [reply lib/Base/Tensor.cpp]: I approved this internally but I was a bit reluctant to push it to OSS.  I noticed that some interaction with buck's test runner as causing llvm::outs() to not be properly flushed when running via `buck test`.  Adding this flush makes the problem go away but it also seems like a workaround for the quirks of an FB-internal tool.  Anyways, if no one minds, I'm happy to have this land.  Just giving the context :)

File: lib/Backends/Habana/Habana.cpp (hunk 1 above)
Comment (arunm-git) [reply lib/Backends/Habana/Habana.cpp]: I can add it. This is the source of truth, so good to add it here.

**Pull request merged**